citar.el

```diff
@@ -413,26 +413,36 @@ documentation for the return value and the meaning of
 REBUILD-CACHE and FILTER."
   (citar-select-ref :rebuild-cache rebuild-cache :multiple t :filter filter))
 
-(defun citar-select-files (files)
-  "Select file(s) from a list of FILES."
-  ;; TODO add links to candidates
-  (completing-read-multiple
-   "Open related file(s): "
-   (lambda (string predicate action)
-     (if (eq action 'metadata)
-         `(metadata
-        ; (group-function . citar-select-group-related-sources)
-           (category . file))
-       (complete-with-action action files string predicate)))))
-
-(defun citar-select-group-related-sources (file transform)
-  "Group by FILE by source, TRANSFORM."
-    (let ((extension (file-name-extension file)))
-      (when transform file
-        ;; Transform for grouping and group title display.
+(defun citar-select-resources (files &optional links)
+  "Select resource(s) from a list of FILES, and optionally LINKS."
+  ;; DONE Add links to candidates
+  ;; DONE Add group-function
+  ;; FIX: set category metadata depending on type (differentiate links and files); possible?
```

Comment:
Whenever I've contemplated something like this (for example with `citar-open`, this has been the hangup.

It is possible and easy to use `consult--multi`, which will do this. But then it's `completing-read` based, so only a single selected candidate.

I simply never figured out how to do it otherwise.

Maybe you can include a screenshot or two on the main post when you get a chance?

Reply:
> It is possible and easy to use `consult--multi`, which will do this. But then it's `completing-read` based, so only a single selected candidate. And it's not explicit dependency.

This is where I landed. Bit of an impasse.